src/main/ruby/truffleruby/core/struct.rb

```diff
@@ -64,7 +64,7 @@ def self.new(klass_name, *attrs, keyword_init: true, &block)
     end
 
     klass = Class.new self do
-      _specialize attrs
+      _specialize attrs unless keyword_init
```

Comment:
Could we keep `_specialize` here and adapt the code for `def initialize` to change based on `keyword_init`?

Reply:
Do you mean add specialization code for `keyword_init`?

Reply:
Yes, I'm thinking we can basically generate a `initialize` like this:
```ruby
def initialize(a: nil,  b: nil)
  TrufflePrimitive.object_hidden_var_set(self, :a, a)
  TrufflePrimitive.object_hidden_var_set(self, :b, b)
end
```
But that's problematic if one of the `attrs` is not a valid local variable name, so it's not trivial indeed.

We could probably do this more easily, and it would still let the Hash accesses be efficient with the given constant key:
```ruby
def initialize(**kwargs) # or def initialize(kwargs = {})
  TrufflePrimitive.object_hidden_var_set(self, kwargs[:a])
  TrufflePrimitive.object_hidden_var_set(self, kwargs[:b])
end
```

> Is it okay if I spend a separate PR for that?

Sounds good.

Reply:
Tracked in https://github.com/oracle/truffleruby/issues/1795